File: lua/hamming/hamming.lua
local hamming = {}

-- compute() returns the Hamming distance between the two DNS strands
function hamming.compute(strand1, strand2)
  if #strand1 ~= #strand2 then return -1 end

  local distance = 0

  for i = 1, #strand1 do
    if strand1:sub(i, i) ~= strand2:sub(i, i) then distance = distance + 1 end
  end

  return distance
end

return hamming

File: ruby/hamming/hamming.rb
# Hamming module to determine distance between two DNA strands
module Hamming
  def self.compute(strand1, strand2)
    raise ArgumentError, 'strand lengths not equal' unless strand1.length == strand2.length

    strand1.chars.zip(strand2.chars).count do |nucleotide1, nucleotide2|
      nucleotide1 != nucleotide2
    end
  end
end

module BookKeeping
  VERSION = 3
end
